airflow/providers/amazon/aws/hooks/dms.py

```diff
@@ -0,0 +1,222 @@
+        """
+        additional_args = create_task_kwargs or {}
+        dms_client = self.get_conn()
+        create_task_response = dms_client.create_replication_task(
+            ReplicationTaskIdentifier=replication_task_id,
+            SourceEndpointArn=source_endpoint_arn,
+            TargetEndpointArn=target_endpoint_arn,
+            ReplicationInstanceArn=replication_instance_arn,
+            MigrationType=migration_type,
+            TableMappings=json.dumps(table_mappings),
+            **additional_args,
+        )
+
+        replication_task_arn = create_task_response['ReplicationTask']['ReplicationTaskArn']
+        self.wait_for_task_status(replication_task_arn, 'ready')
+
+        return replication_task_arn
+
+    def start_replication_task(
+        self,
+        replication_task_arn: str,
+        start_replication_task_type: str,
+        start_task_kwargs: Optional[dict] = None,
+    ):
+        """
+        Starts replication task.
+
+        :param replication_task_arn: Replication task ARN
+        :type replication_task_arn: str
+        :param start_replication_task_type: Replication task start type
+            ('start-replication'|'resume-processing'|'reload-target')
+        :type start_replication_task_type: str
+        :param start_task_kwargs: Extra start replication task arguments
+        :type start_task_kwargs: Optional[dict]
+        """
+        additional_args = start_task_kwargs or {}
+        dms_client = self.get_conn()
+        dms_client.start_replication_task(
+            ReplicationTaskArn=replication_task_arn,
+            StartReplicationTaskType=start_replication_task_type,
+            **additional_args,
+        )
+
+    def stop_replication_task(self, replication_task_arn):
+        """
+        Stops replication task.
+
+        :param replication_task_arn: Replication task ARN
+        :type replication_task_arn: str
+        """
+        dms_client = self.get_conn()
+        dms_client.stop_replication_task(ReplicationTaskArn=replication_task_arn)
+
+    def delete_replication_task(self, replication_task_arn):
+        """
+        Starts replication task deletion and waits for it to be deleted
+
+        :param replication_task_arn: Replication task ARN
+        :type replication_task_arn: str
+        """
+        dms_client = self.get_conn()
+        dms_client.delete_replication_task(ReplicationTaskArn=replication_task_arn)
+
+        self.wait_for_task_status(replication_task_arn, 'deleted')
+
+    def wait_for_task_status(self, replication_task_arn: str, status: str):
+        """
+        Waits for replication task to reach status.
+        Supported statuses: deleted, ready, running, stopped.
+
+        :param status: Status to wait for
+        :type status: str
+        :param replication_task_arn: Replication task ARN
+        :type replication_task_arn: str
+        """
+        available_statuses = ['deleted', 'ready', 'running', 'stopped']
```

Comment:
Small nit: you could move this to an Enum instead of a local variable. Also that way the method stub would be more strongly typed as well.